datafusion/functions-aggregate/src/string_agg.rs

```diff
@@ -108,15 +108,14 @@ impl AggregateUDFImpl for StringAgg {
 
     fn accumulator(&self, acc_args: AccumulatorArgs) -> Result<Box<dyn Accumulator>> {
         if let Some(lit) = acc_args.exprs[1].as_any().downcast_ref::<Literal>() {
-            return match lit.value() {
-                ScalarValue::Utf8(Some(delimiter))
-                | ScalarValue::LargeUtf8(Some(delimiter)) => {
-                    Ok(Box::new(StringAggAccumulator::new(delimiter.as_str())))
+            return match lit.value().try_as_str() {
```

Comment:
This is a pretty good example of  can reducing the repetition in the code to check for string literal values.  This also now implicitly will work for Dictionary values where it would not have before